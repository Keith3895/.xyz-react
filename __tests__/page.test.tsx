import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../src/app/page';

describe('Home', () => {
  it('renders without crashing', () => {
    const { container } = render(<Home />);
    expect(container).toBeInTheDocument();
  });

  it('renders expected text', () => {
    render(<Home />);
    expect(screen.getByText('Hello')).toBeInTheDocument();
  });
  it('renders homepage unchanged', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  });
});