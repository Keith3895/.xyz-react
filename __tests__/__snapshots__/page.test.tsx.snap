// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div>
    <div
      class="flex w-full bgframe justify-center items-center max-md:flex-col-reverse max-md:mb-4"
    >
      <div
        class="lg:w-1/2 lg:h-3/5  flex flex-col justify-center items-center"
      >
        <div>
          <h1
            class="text-4xl font-bold mt-20 md:ml-16 ml-4"
          >
            Hello
            <span
              class="text-custom-orange-600 text-4xl font-bold"
            >
              .
            </span>
          </h1>
          <h3
            class="text-4xl font-bold mt-4 md:ml-24 ml-8"
          >
            I
            <span
              class="text-custom-orange-600 text-4xl font-bold"
            >
              ’
            </span>
            m Keith Franklin
          </h3>
          <h1
            class="text-4xl font-bold mt-4 md:ml-20 ml-6"
          >
            Tech Enthusiast
            <span
              class="text-custom-orange-600 text-4xl font-bold"
            >
              .
            </span>
          </h1>
          <div
            class="flex flex-row  md:ml-16 ml-4 mt-14"
          >
            <button
              class="w-36 h-10 bg-custom-orange-600  justify-center items-center inline-flex cursor-pointer hover:bg-custom-orange-400 active:bg-custom-orange-700"
              type="button"
            >
              <div
                class="text-center text-black text-base font-bold "
              >
                Contact me
              </div>
            </button>
            <a
              class="w-36 h-10 bg-white border-4 border-custom-orange-600 justify-center items-center cursor-pointer inline-flex hover:bg-grey active:bg-custom-orange-100 ml-4"
              download="Resume"
              href="./Resume.pdf"
            >
              <div
                class="text-center text-black text-base font-bold "
              >
                My Résumé
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="max-md:bg-black lg:w-60 lg:h-1/2 flex items-center justify-center"
      >
        <img
          alt=""
          class=" md:mt-14 h-auto w-112"
          data-nimg="1"
          decoding="async"
          height="500"
          loading="lazy"
          src="/_next/image?url=%2Fhero%20image.png&w=1080&q=100"
          srcset="/_next/image?url=%2Fhero%20image.png&w=640&q=100 1x, /_next/image?url=%2Fhero%20image.png&w=1080&q=100 2x"
          style="color: transparent;"
          width="500"
        />
      </div>
    </div>
    <div
      class="relative flex overflow-x-hidden bg-black bg-opacity-90"
    >
      <div
        class="py-5 animate-marquee whitespace-nowrap"
      >
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          Angular
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          React
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          Flutter
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          NodeJS
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          Python
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          web3
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          MongoDB
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          PostgreSQL
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          MySql
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          Kubernetes
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          docker
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          helm
        </span>
      </div>
      <div
        class="absolute top-0 py-5 animate-marquee2 whitespace-nowrap"
      >
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          Angular
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          React
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          Flutter
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          NodeJS
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          Python
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          web3
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          MongoDB
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          PostgreSQL
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          MySql
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          Kubernetes
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          docker
        </span>
        <span
          class="uppercase text-center text-white text-opacity-40 text-3xl font-bold mx-4"
        >
          helm
        </span>
      </div>
    </div>
    <div
      class="flex flex-col w-full bg-tertiary-600 px-12 py-4 space-y-4"
    >
      <h2
        class="text-4xl font-bold flex"
      >
        Recent Activity
      </h2>
      <ol
        class="items-center sm:flex"
      >
        <li
          class="relative mb-6 sm:mb-0"
        >
          <div
            class="flex items-center"
          >
            <div
              class="z-10 flex items-center justify-center w-6 h-6 bg-blue-100 rounded-full ring-0 ring-white dark:bg-blue-900 sm:ring-8 dark:ring-gray-900 shrink-0"
            >
              <svg
                aria-hidden="true"
                class="w-2.5 h-2.5 text-blue-800 dark:text-blue-300"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 4a2 2 0 0 0-2-2h-2V1a1 1 0 0 0-2 0v1h-3V1a1 1 0 0 0-2 0v1H6V1a1 1 0 0 0-2 0v1H2a2 2 0 0 0-2 2v2h20V4ZM0 18a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V8H0v10Zm5-8h10a1 1 0 0 1 0 2H5a1 1 0 0 1 0-2Z"
                />
              </svg>
            </div>
            <div
              class="hidden sm:flex w-full bg-gray-200 h-0.5 dark:bg-gray-700"
            />
          </div>
          <div
            class="mt-3 sm:pe-8"
          >
            <h3
              class="text-lg font-semibold text-custom-orange-600"
            >
              What is web 3.0?
            </h3>
            <time
              class="block mb-2 text-sm font-normal leading-none text-white"
            >
              Upcoming
            </time>
            <p
              class="text-base font-normal text-gray-400"
            >
              Navigationg the jargon and trying to unpack what is web3.
            </p>
          </div>
        </li>
        <li
          class="relative mb-6 sm:mb-0"
        >
          <div
            class="flex items-center"
          >
            <div
              class="z-10 flex items-center justify-center w-6 h-6 bg-blue-100 rounded-full ring-0 ring-white dark:bg-blue-900 sm:ring-8 dark:ring-gray-900 shrink-0"
            >
              <svg
                aria-hidden="true"
                class="w-2.5 h-2.5 text-blue-800 dark:text-blue-300"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 4a2 2 0 0 0-2-2h-2V1a1 1 0 0 0-2 0v1h-3V1a1 1 0 0 0-2 0v1H6V1a1 1 0 0 0-2 0v1H2a2 2 0 0 0-2 2v2h20V4ZM0 18a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V8H0v10Zm5-8h10a1 1 0 0 1 0 2H5a1 1 0 0 1 0-2Z"
                />
              </svg>
            </div>
            <div
              class="hidden sm:flex w-full bg-gray-200 h-0.5 dark:bg-gray-700"
            />
          </div>
          <div
            class="mt-3 sm:pe-8"
          >
            <h3
              class="text-lg font-semibold text-custom-orange-600"
            >
              Car wash app
            </h3>
            <time
              class="block mb-2 text-sm font-normal leading-none text-white"
            >
              In Progress
            </time>
            <p
              class="text-base font-normal text-gray-400"
            >
              A car wash service aggregator app.
            </p>
          </div>
        </li>
        <li
          class="relative mb-6 sm:mb-0"
        >
          <div
            class="flex items-center"
          >
            <div
              class="z-10 flex items-center justify-center w-6 h-6 bg-blue-100 rounded-full ring-0 ring-white dark:bg-blue-900 sm:ring-8 dark:ring-gray-900 shrink-0"
            >
              <svg
                aria-hidden="true"
                class="w-2.5 h-2.5 text-blue-800 dark:text-blue-300"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 4a2 2 0 0 0-2-2h-2V1a1 1 0 0 0-2 0v1h-3V1a1 1 0 0 0-2 0v1H6V1a1 1 0 0 0-2 0v1H2a2 2 0 0 0-2 2v2h20V4ZM0 18a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V8H0v10Zm5-8h10a1 1 0 0 1 0 2H5a1 1 0 0 1 0-2Z"
                />
              </svg>
            </div>
            <div
              class="hidden sm:flex w-full bg-gray-200 h-0.5 dark:bg-gray-700"
            />
          </div>
          <div
            class="mt-3 sm:pe-8"
          >
            <h3
              class="text-lg font-semibold text-custom-orange-600"
            >
              Rewamp of my portfolio
            </h3>
            <time
              class="block mb-2 text-sm font-normal leading-none text-white"
            >
              Released on 2024-02-15
            </time>
            <p
              class="text-base font-normal text-gray-400"
            >
              updated my portfolio with new projects and technologies. And a complete 
                            design overhaul.
            </p>
          </div>
        </li>
      </ol>
    </div>
    <div
      class="flex flex-col p-10 justify-center items-center space-y-10"
    >
      <div
        class="flex flex-row justify-center"
      >
        <div
          class="w-96 h-20 bg-white border-8 border-custom-orange-600 justify-center items-center inline-flex"
        >
          <div
            class="w-96 text-center text-black text-3xl font-bold  leading-10 tracking-widest"
          >
            ABOUT ME
          </div>
        </div>
      </div>
      <div
        class="w-full md:w-3/5 text-left text-base font-normal  leading-loose"
      >
        <p>
          Hey there! I'm fueled by a relentless curiosity for all things tech. Currently, I'm proudly steering the tech ship as a Technical Lead at 
          <a
            class="hover:underline after:content-['_↗']"
            href="https://auberginesolutions.com/"
            rel="noreferrer"
            target="_blank"
          >
            Aubergine
          </a>
          , where every day is an adventure in innovation.
        </p>
        <p>
          With 
          6
          + years of immersion in the tech realm under my belt, I've become somewhat of a tech explorer. From tinkering with new applications to diving headfirst into the latest tech trends, I'm always on the lookout for the next big thing.
        </p>
        <p>
          Beyond my professional pursuits, I find solace in the melodic world of music. Dreaming of owning a record player and indulging in the crisp sounds of my favourite tunes is a personal goal I'm eagerly working towards.
        </p>
        <p>
          If you're as jazzed about tech and innovation as I am, let's connect! Swing by my 
          <a
            class="hover:underline after:content-['_↗']"
            href="https://github.com/Keith3895"
            rel="noreferrer"
            target="_blank"
          >
            GitHub
          </a>
           to see what I've been cooking up lately, or drop me a message on 
          <a
            class="hover:underline after:content-['_↗']"
            href="mailto:contact.keithfranklin@gmail.com"
            rel="noreferrer"
            target="_blank"
          >
            link
          </a>
           for a chat about all things tech (or music, I'm game for that too!).
        </p>
        Here's to endless exploration and innovation!
      </div>
      <div
        class="flex flex-col w-full justify-center items-center lg:pt-20 md:pt-20"
      >
        <div
          class="w-3/4 flex flex-row"
        >
          <div
            class="w-1/12 md:w-1/3 h-60 px-5 pt-24 pb-11 flex-col justify-end items-center inline-flex"
          />
          <div
            class="w-4/5 md:w-1/3 h-60 px-5 bg-custom-orange-600 flex-col justify-center items-center inline-flex"
          >
            <div
              class="text-center"
            >
              <span
                class="text-black text-4xl font-bold  leading-10"
              >
                6
              </span>
              <span
                class="text-black text-4xl font-bold font-['Poppins'] leading-10"
              >
                +
              </span>
            </div>
            <div
              class=" text-center text-black text-xl font-bold  leading-7"
            >
              Years experience
            </div>
          </div>
          <div
            class=" md:w-1/3 h-60 px-5 pt-24 pb-11 flex-col justify-end items-center inline-flex"
          />
        </div>
        <div
          class="w-full md:w-3/4 flex flex-row"
        >
          <div
            class="w-1/2 md:w-1/3 h-60 px-5 bg-black flex-col justify-center items-center inline-flex"
          >
            <div
              class="text-center"
            >
              <span
                class="text-white text-4xl font-bold  leading-10"
              >
                20
              </span>
              <span
                class="text-white text-4xl font-bold font-['Poppins'] leading-10"
              >
                +
              </span>
            </div>
            <div
              class=" text-center text-white text-xl font-bold  leading-7"
            >
              Projects worked on
            </div>
          </div>
          <div
            class="hidden md:flex md:w-1/3 h-60 px-5 pt-24 pb-11 flex-col justify-end items-center inline-flex"
          />
          <div
            class="w-1/2 md:w-1/3 h-60 px-5 bg-tertiary-600 flex-col justify-center items-center inline-flex"
          >
            <div
              class="text-center"
            >
              <span
                class="text-white text-4xl font-bold  leading-10"
              >
                10
              </span>
              <span
                class="text-white text-4xl font-bold font-['Poppins'] leading-10"
              >
                +
              </span>
            </div>
            <div
              class=" text-center text-white text-xl font-bold  leading-7"
            >
              Clients worked with
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
